libsql.go

```diff
@@ -772,9 +786,37 @@ Outerloop:
 }
 
 func (c *conn) QueryContext(ctx context.Context, query string, args []sqldriver.NamedValue) (sqldriver.Rows, error) {
-	rowsNativePtr, err := c.execute(query, args, false)
+	rowsNativePtr, err := c.executeContext(ctx, query, args, false)
 	if err != nil {
 		return nil, err
 	}
 	return newRows(rowsNativePtr)
 }
+
+// doContext runs a blocking function f in a goroutine, and exits early if the given context is
+// canceled or if its deadline is exceeded.
+func doContext[U any](ctx context.Context, f func() (U, error)) (U, error) {
+	var res U
+	resCh := make(chan U)
+	errCh := make(chan error)
+
+	go func() {
+		defer close(resCh)
+		defer close(errCh)
+		res, err := f()
```

Comment:
How is this safe? If context times out, why is it OK to let f() still run? Isn't the problem here that maybe `f()` actually succeeds or does something in the background, while we think it already failed because we return after context timeout. We should probably just pass `context` to `f()`?